app/src/main/java/app/revanced/manager/compose/ui/viewmodel/MainViewModel.kt

```diff
@@ -0,0 +1,25 @@
+package app.revanced.manager.compose.ui.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import app.revanced.manager.compose.domain.repository.SourceRepository
+import app.revanced.manager.compose.util.PM
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class MainViewModel(
+    sourceRepository: SourceRepository,
+    pm: PM
+) : ViewModel() {
+    init {
+        viewModelScope.launch {
+            sourceRepository.loadSources()
+        }
+        viewModelScope.launch {
+            pm.getCompatibleApps()
+        }
+        viewModelScope.launch(Dispatchers.IO) {
+            pm.getInstalledApps()
+        }
```

Comment:
Default probably makes more sense for `getInstalledApps()`.

```suggestion
viewModelScope.launch {
        launch {
            sourceRepository.loadSources()
        }
        launch {
            pm.getCompatibleApps()
        }
        launch(Dispatchers.Default) {
            pm.getInstalledApps()
        }
}
```

Reply:
No, loading from the packageManager should be done from the IO thread according to canny

Reply:
Oh ok. Nevermind that then.
(you should still consider putting all the launches inside a single `viewModelScope.launch` or `with` block)